yt/python/yt/cli/strawberry_parser.py

```diff
@@ -0,0 +1,157 @@
+                if env_variable:
+                    # We add family prefix for all ctl-related env variables to prevent stealing
+                    # sensitive env variables if controller is compromised.
+                    default_value = os.getenv(family_upper + "_" + env_variable)
+                    description_ext = "default value can be set via {}_{} env variable".format(family_upper,
+                                                                                               env_variable)
+
+                    if default_value is not None:
+                        description_ext += " (current value: \"{}\")".format(default_value)
+                        # If there is a default value, the parameter is not required anymore.
+                        required = False
+
+                    if description:
+                        description += "; " + description_ext
+                    else:
+                        description = description_ext
+
+                element_name = param.get("element_name")
+                element_description = param.get("element_description")
+                element_aliases = [("-" if len(a) == 1 else "--") + a.replace("_", "-")
+                                   for a in param.get("element_aliases", [])]
+
+                if as_positional_argument:
+                    add_argument(
+                        subparser,
+                        name,
+                        description=description)
+                elif element_name:
+                    group = subparser.add_mutually_exclusive_group()
+                    add_argument(
+                        group,
+                        "--" + name.replace("_", "-"),
+                        action=param.get("action"),
+                        description=description,
+                        aliases=aliases)
+                    add_argument(
+                        group,
+                        "--" + element_name.replace("_", "-"),
+                        action="append",
+                        aliases=element_aliases,
+                        dest=name,
+                        description=element_description)
+                else:
+                    add_argument(
+                        subparser,
+                        "--" + name.replace("_", "-"),
+                        action=param.get("action"),
+                        description=description,
+                        required=required,
+                        aliases=aliases,
+                        default=default_value)
+
+    # We replace original "parse_known_args" with our implementation, which fetches and registers available
+    # commands before parsing.
+    do_parse_known_args = parser.parse_known_args
+
+    def parse_known_args(*args, **kwargs):
+        parsed_address, unparsed = address_parser.parse_known_args(*args, **kwargs)
+        register_commands(parsed_address.address)
+        return do_parse_known_args(*args, **kwargs)
```

Comment:
Only unparsed arguments should be propagated to `do_parse_known_args`.

It's not your bug, but let's fix it :)